source/core/errors.ts

```diff
@@ -170,3 +170,13 @@ export class RetryError extends RequestError {
 		this.code = 'ERR_RETRYING';
 	}
 }
+
+/**
+An error to be thrown when the request is aborted by AbortController.
+*/
+export class AbortError extends RequestError {
+	constructor(request: Request, error?: Error) {
+		super('This operation was aborted.', {...error, code: 'ERR_ABORTED'}, request);
```

Comment:
Since `error` (abortsignal.reason) is optional here, it seems passing `error` into the second argument of the constructor cause compile error.

```
// Compile error occurs

// Argument of type 'ErrnoException | undefined' is not assignable to parameter of type 'Partial<ErrnoException & { code?: string | undefined; }>'.
// Type 'undefined' is not assignable to type 'Partial<ErrnoException & { code?: string | undefined; }>'
super('This operation was aborted.', error, request);
this.code = 'ERR_ABORTED';
```

Reply:
I see, thanks for clarifying. In this case you can just skip this. Like so:

```js
super('This operation was aborted.', {}, request);
this.name = 'AbortError';
this.code = 'ERR_ABORTED';
```

The reason is user accessible via `controller.signal.reason`, so I don't think we need to expose it for now. That might be for https://github.com/sindresorhus/got/issues/1953